src/cache_cf.cc

```diff
@@ -4367,20 +4443,22 @@ static void parse_icap_service_failure_limit(Adaptation::Icap::Config *cfg)
 
     d = static_cast<time_t> (xatoi(token));
 
-    m = static_cast<time_t> (1);
-
+    using Seconds = std::chrono::seconds;
+    std::chrono::nanoseconds parsedUnitDuration(Seconds(1));
     if (0 == d)
         (void) 0;
     else if ((token = ConfigParser::NextToken()) == NULL) {
         debugs(3, DBG_CRITICAL, "No time-units on '" << config_input_line << "'");
         self_destruct();
         return;
-    } else if ((m = parseTimeUnits(token, false)) == 0) {
+    } else if (!parseTimeUnits<Seconds>(token, nullptr, parsedUnitDuration)) {
         self_destruct();
         return;
     }
 
-    cfg->oldest_service_failure = (m * d);
+    CheckTimeValue(d, parsedUnitDuration);
```

Comment:
It feels odd that we are calling CheckTimeValue() (which checks, among other things, for nanosecond storage overflows) when oldest_service_failure does not require a nanosecond precision. Is this because we want any possible future conversion of `oldest_service_failure` to nanoseconds to be safe? Even if there is no such conversion in the current code? If yes, we should adjust CheckTimeValue() code to add a comment that explains this design decision.